Ktisis/Interface/ConfigGui.cs

```diff
@@ -83,16 +91,77 @@ internal class ConfigGui {
 
 		public void DrawOverlayTab() {
 			var drawLines = Cfg.DrawLinesOnSkeleton;
-			var lineThickness = Cfg.SkeletonLineThickness;
 			if (ImGui.Checkbox("Draw lines on skeleton", ref drawLines)) {
 				Cfg.DrawLinesOnSkeleton = drawLines;
 				Cfg.Save(Plugin);
 			}
-			if (ImGui.SliderFloat("Lines thickness", ref lineThickness,0.01F , 15F, "%.1f")) {
+
+			var lineThickness = Cfg.SkeletonLineThickness;
+			if (ImGui.SliderFloat("Lines thickness", ref lineThickness, 0.01F, 15F, "%.1f")) {
 				Cfg.SkeletonLineThickness = lineThickness;
 				Cfg.Save(Plugin);
 			}
 
+			ImGui.Separator();
+
+			bool linkBoneCategoriesColors = Cfg.LinkBoneCategoryColors;
+			if (ImGuiComponents.IconButton(FontAwesomeIcon.Link, linkBoneCategoriesColors ? new Vector4(0.0F, 1.0F, 0.0F, 0.4F) : null))
+			{
+				Cfg.LinkBoneCategoryColors = !linkBoneCategoriesColors;
+				Cfg.Save(Plugin);
+			}
+
+			ImGui.SameLine();
+			ImGui.Text(linkBoneCategoriesColors ? "Unlink bones colors" : "Link bones colors");
+
+			ImGui.SameLine();
+			if (ImGuiComponents.IconButton(FontAwesomeIcon.Eraser))
+			{
+				Vector4 eraseColor = new Vector4(1.0F, 1.0F, 1.0F, 0.5647059F);
+				if(linkBoneCategoriesColors)
+					Cfg.LinkedBoneCategoryColor = eraseColor;
+				else
+					foreach (Category category in BoneCategories.Categories)
+						Cfg.BoneCategoryColors[category] = eraseColor;
+				Cfg.Save(Plugin);
+			}
+
+
+			if (linkBoneCategoriesColors)
+			{
+				Vector4 linkedBoneColor = Cfg.LinkedBoneCategoryColor;
+				if (ImGui.ColorEdit4("Bones color", ref linkedBoneColor, ImGuiColorEditFlags.NoInputs | ImGuiColorEditFlags.AlphaBar))
+				{
+					Cfg.LinkedBoneCategoryColor = linkedBoneColor;
+					Cfg.Save(Plugin);
+				}
+
+			} else {
+
+				ImGui.SameLine();
+				if (ImGuiComponents.IconButton(FontAwesomeIcon.Rainbow))
+				{
+					foreach (Category category in BoneCategories.Categories)
+					{
+						Cfg.BoneCategoryColors[category] = category.DefaultColor;
+					}
+					Cfg.Save(Plugin);
+				}
+
+				ImGui.Text("Bone colors by category");
+
+				foreach (Category category in BoneCategories.Categories)
+				{
+					Vector4 categoryColor = new Vector4(1.0F, 1.0F, 1.0F, 0.5647059F);
+					if (Cfg.BoneCategoryColors.ContainsKey(category)) categoryColor = Cfg.BoneCategoryColors[category];
```

Comment:
Can you refactor this to `TryGetValue`?

Reply:
Actually I can't, it seems `TryGetValue` erases the default when it's unable to find the key.
So the color is `#00000000`.

Reply:
It also returns `false`, so you can just set it to the correct color in that case.

Reply:
Ah! it occurred to me that could be the case, but thought it wouldn't, since it goes fine in the `Vector4` type.

Reply:
i had to convert some of the dictionaries to nullable type for this, but I think it is still cleaner than before.
Also I suppose it is better for performance.